src/libraries/Common/tests/TestUtilities/System/WindowsIdentityFixture.cs

```diff
@@ -118,8 +118,8 @@ private void CreateUser()
         [return: MarshalAs(UnmanagedType.Bool)]
         private static partial bool LogonUser(string userName, string domain, string password, int logonType, int logonProvider, out SafeAccessTokenHandle safeAccessTokenHandle);
 
-        [LibraryImport("netapi32.dll", SetLastError = true)]
-        internal static partial uint NetUserAdd([MarshalAs(UnmanagedType.LPWStr)] string servername, uint level, ref USER_INFO_1 buf, out uint parm_err);
+        [DllImport("netapi32.dll", SetLastError = true)]
```

Comment:
@AaronRobinsonMSFT do you know what the issue is?

Reply:
That is odd. What configuration are we building with here?

/cc @dotnet/interop-contrib

Reply:
Also, this file looks like it should only compile on Windows. Are we missing a guard somewhere?

Reply:
> That is odd. What configuration are we building with here?

It's not a CI configuration.

I'm running a build similar to that of source-build **with tests** using this command:
```
./build.sh --runtimeconfiguration Release --librariesConfiguration Debug --subset clr+libs+libs.tests --test /p:DotNetBuildFromSource=true /p:RuntimeOS=linux
```

> Also, this file looks like it should only compile on Windows. Are we missing a guard somewhere?

From this comment, platforms are detected at runtime for this file:

https://github.com/dotnet/runtime/blob/873775b42431bf27a9ae69633864e5290443c29b/src/libraries/Common/tests/TestUtilities/TestUtilities.csproj#L34-L38

Are there patterns `LibraryImport` can only handle on Windows, or is it platform agnostic?

Reply:
> Are there patterns LibraryImport can only handle on Windows, or is it fully platform agnostic?

It should be fully platform agnostic - at least with respect to the source gen.

Reply:
Maybe we're using a different version of the source generator under this configuration.

How/where does the source generator get referenced?

Reply:
The source generator is defined at https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/LibraryImportGenerator.csproj.

I think it is chained in via the below property, but I'm not entirely sure. That is a question for @elinor-fung or @jkoritzinsky.

https://github.com/dotnet/runtime/blob/ff2a37807ed9fd99e4a90cda2b28292e1f5d73c6/src/libraries/NetCoreAppLibrary.props#L185-L186

Reply:
> [/home/tmds/repos/runtime/src/libraries/Common/tests/TestUtilities/TestUtilities.csproj::TargetFramework=net8.0]

I think this is actually because `NetCoreAppMinimum` is `net8.0` in source build, but `net6.0` in non-source build.

https://github.com/dotnet/runtime/blob/73f15736110a0236d0005e22ad7c770ee6e13ab9/Directory.Build.props#L71-L72

When targeting net6.0, we do a downlevel fallback of just emitting `DllImport`. For net7.0+, it will try to generate marshalling code, but there isn't any marshalling info for `USER_INFO_1`, so the generator errors:
https://github.com/dotnet/runtime/blob/73f15736110a0236d0005e22ad7c770ee6e13ab9/src/libraries/Common/tests/TestUtilities/System/WindowsIdentityFixture.cs#L127-L129

Reply:
@elinor-fung then it's appropriate to change this to a `DllImport` to make the code compile for `net8.0`?

Reply:
> @elinor-fung then it's appropriate to change this to a `DllImport` to make the code compile for `net8.0`?

That is one option. The other is to write a custom struct marshaller. For now, it seems appropriate to switch to `DllImport` and file an issue so we can author a custom marshaller.

Related ask: https://github.com/dotnet/runtime/issues/81656

Reply:
Do you want me to write an issue for this, or will you do it?

Reply:
Filed https://github.com/dotnet/runtime/issues/81705